### PR TITLE
Add SelectionBrush and SelectionForegroundBrush feature to TextBox

### PR DESCRIPTION
**Pull request by Abdulbaqi-Alshareef**
## What does the pull request do?
This PR adds SelectionBrush styled property to TextBox to change the color(brush) of the selected text(Highlight color) and SelectionForegroundBrush  to change the color of the foreground of it

CaretSelection

Also adds example to show these features in Control Catalog

## What is the current behavior?
Now selection brush can only be changed by overriding the default theme key "HighlightColor" ,
and you can't change the foreground selected text which is white by default.

## What is the updated/expected behavior with this PR?
Selection and foreground selection colors can be changed by setting color name or brush of SelectionBrush and SelectionForegroundBrush  styled properties like this:

`<TextBox   Text="Custom selection brush" SelectionStart="5" SelectionEnd="22" SelectionBrush="Green" SelectionForegroundBrush="Yellow"/>`

## Breaking changes
I think there is no Breaking changes

## Fixed issues

Adds feature requested here #2628
With this  PR and #2651 closes this request #2628

### CODE REVIEW

File: src/Avalonia.Controls/Presenters/TextPresenter.cs [hunk 1]
```diff
@@ -129,14 +147,11 @@ public override void Render(DrawingContext context)
 
                 var rects = FormattedText.HitTestTextRange(start, length);
 
-                if (_highlightBrush == null)
-                {
-                    _highlightBrush = (IBrush)this.FindResource("HighlightBrush");
-                }
+                _selectionBrush = SelectionBrush;
```
Comment (Gillibald) [inline src/Avalonia.Controls/Presenters/TextPresenter.cs]: I think this can be a local variable.

File: src/Avalonia.Themes.Default/TextBox.xaml [hunk 2]
```diff
@@ -3,6 +3,7 @@
     <Setter Property="Background" Value="{DynamicResource ThemeBackgroundBrush}"/>
     <Setter Property="BorderBrush" Value="{DynamicResource ThemeBorderMidBrush}"/>
     <Setter Property="BorderThickness" Value="{DynamicResource ThemeBorderThickness}"/>
+    <Setter Property="SelectionBrush" Value="{DynamicResource HighlightBrush}"/>
```
Comment (Gillibald) [inline src/Avalonia.Themes.Default/TextBox.xaml]: The default for SelectionForegroundBrush can be defined here as well.

File: src/Avalonia.Controls/Presenters/TextPresenter.cs [hunk 3]
```diff
@@ -247,12 +262,13 @@ protected override FormattedText CreateFormattedText(Size constraint, string tex
             var selectionEnd = SelectionEnd;
             var start = Math.Min(selectionStart, selectionEnd);
             var length = Math.Max(selectionStart, selectionEnd) - start;
+            var selectionForegroundBrush = SelectionForegroundBrush ?? Brushes.White;
```
Comment (Gillibald) [inline src/Avalonia.Controls/Presenters/TextPresenter.cs]: This default can be defined in the default style.

File: src/Avalonia.Controls/Presenters/TextPresenter.cs (hunk 1 above)
Comment (Abdulbaqi-Alshareef) [reply src/Avalonia.Controls/Presenters/TextPresenter.cs]: Could you explain more?

Comment (Gillibald) [reply src/Avalonia.Controls/Presenters/TextPresenter.cs]: just change it to `var selectionBrush = SelectionBrush` and remove `_selectionBrush`

File: src/Avalonia.Themes.Default/TextBox.xaml (hunk 2 above)
Comment (Abdulbaqi-Alshareef) [reply src/Avalonia.Themes.Default/TextBox.xaml]: Yes I think so and it needs to define the default brush in default theme may be as "HighlightForegroundBrush"

File: src/Avalonia.Controls/Presenters/TextPresenter.cs (hunk 1 above)
Comment (Abdulbaqi-Alshareef) [reply src/Avalonia.Controls/Presenters/TextPresenter.cs]: Yes , or maybe remove _selectionBrush and just use SelectionBrush styled property?

File: src/Avalonia.Controls/Presenters/TextPresenter.cs [hunk 4]
```diff
@@ -19,6 +19,12 @@ public class TextPresenter : TextBlock
         public static readonly StyledProperty<char> PasswordCharProperty =
             AvaloniaProperty.Register<TextPresenter, char>(nameof(PasswordChar));
 
+        public static readonly StyledProperty<IBrush> SelectionBrushProperty =
+           AvaloniaProperty.Register<TextPresenter, IBrush>(nameof(SelectionBrushProperty));
+
+        public static readonly StyledProperty<IBrush> SelectionForegroundBrushProperty =
+        AvaloniaProperty.Register<TextPresenter, IBrush>(nameof(SelectionForegroundBrushProperty));
```
Comment (kekekeks) [inline src/Avalonia.Controls/Presenters/TextPresenter.cs]: indent

File: src/Avalonia.Controls/TextBox.cs [hunk 5]
```diff
@@ -38,6 +38,12 @@ public class TextBox : TemplatedControl, UndoRedoHelper<TextBox.UndoRedoState>.I
         public static readonly StyledProperty<char> PasswordCharProperty =
             AvaloniaProperty.Register<TextBox, char>(nameof(PasswordChar));
 
+        public static readonly StyledProperty<IBrush> SelectionBrushProperty =
+           AvaloniaProperty.Register<TextBox, IBrush>(nameof(SelectionBrushProperty));
+
+        public static readonly StyledProperty<IBrush> SelectionForegroundBrushProperty =
+        AvaloniaProperty.Register<TextBox, IBrush>(nameof(SelectionForegroundBrushProperty));
```
Comment (kekekeks) [inline src/Avalonia.Controls/TextBox.cs]: indent

**Review by kekekeks — APPROVED** [review]
LGTM

File: src/Avalonia.Controls/Presenters/TextPresenter.cs (hunk 4 above)
Comment (Abdulbaqi-Alshareef) [reply src/Avalonia.Controls/Presenters/TextPresenter.cs]: Fixed

File: src/Avalonia.Controls/TextBox.cs (hunk 5 above)
Comment (Abdulbaqi-Alshareef) [reply src/Avalonia.Controls/TextBox.cs]: Fixed

**Pull request merged**